pkg/synchronization/core/ignore_test.go

```diff
@@ -61,24 +79,25 @@ func TestIgnorerBasic(t *testing.T) {
 			"somedir/",
 		},
 		tests: []ignoreTestValue{
-			{"", false, false},
-			{"", true, false},
-			{"something", false, false},
-			{"something", true, false},
-			{"subpath/something", false, false},
-			{"subpath/something", true, false},
-			{"otherthing", false, true},
-			{"otherthing", true, true},
-			{"subpath/otherthing", false, true},
-			{"subpath/otherthing", true, true},
-			{"random", false, false},
-			{"random", true, false},
-			{"subpath/random", false, false},
-			{"subpath/random", true, false},
-			{"somedir", false, false},
-			{"somedir", true, true},
-			{"subpath/somedir", false, false},
-			{"subpath/somedir", true, true},
+			{"", false, false, false},
+			{"", true, false, false},
+			{"something", false, false, false},
+			{"something", true, false, false},
+			{"otherthing", false, true, true},
+			{"otherthing", true, true, true},
+			{"random", false, false, false},
+			{"random", true, false, false},
+			{"somedir", false, false, true},
+			{"somedir", true, true, true},
+			{"subpath", true, false, false},
+			{"subpath/something", false, false, false},
```

Comment:
Note I re-ordered many of these tests so that "subpath" is inspected before "subpath/..." because the `fileutils.MatchesUsingParentResults` assumes depth first traversal to use parent context for the doublestar glob matching.